src/renderer/redux/selectors/file_info.js

```diff
@@ -103,3 +105,91 @@ export const selectTotalDownloadProgress = createSelector(selectDownloadingFileI
   if (fileInfos.length > 0) return totalProgress / fileInfos.length / 100.0;
   return -1;
 });
+
+export const selectSearchDownloadUris = (query) => createSelector(
+  selectFileInfosDownloaded,
+  selectClaimsById,
+  (fileInfos, claimsById) => {
+    if (!query || !fileInfos.length) {
+      return null;
+    }
+
+    const queryParts = query.toLowerCase().split(' ');
+    const searchQueryDictionary = {};
+    queryParts.forEach(subQuery => {
+      searchQueryDictionary[subQuery] = subQuery;
+    });
+
+    const arrayContainsQueryPart = (array) => {
+      for (var i = 0; i < array.length; i += 1) {
+        const subQuery = array[i];
+        if (searchQueryDictionary[subQuery]) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    let downloadResultsFromQuery = [];
+    fileInfos.forEach((fileInfo) => {
+      const { channel_name, name, metadata } = fileInfo;
+      const { stream: { metadata: { author, description, title } } } = metadata;
+
+      if (channel_name) {
+        const channelName = channel_name.toLowerCase();
+        const strippedOutChannelName = channelName.slice(1); // trim off the @
```

Comment:
since im talking about regex a lot in this code review

I think it is better to do this by replace `/^@/` (`@` at the beginning of the string) with `''` empty string, instead of truncating the first letter. In that case, it wouldn't break if the channel format changed to not include an @.

Not important though